utils/utils.py

```diff
@@ -247,17 +253,17 @@ def expand_braces(pattern: str):
 def files_names_from_pattern(files_pattern):
     data_files_names = None
     path_expander = glob.glob
-    if not no_deps_mode:
-        from braceexpand import braceexpand
+    data_files_names = (
+        list(braceexpand(files_pattern)) if braceexpand else expand_braces(files_pattern)
+    )
 
-        data_files_names = list(braceexpand(files_pattern))
-        if "://" in files_pattern:
-            if all(map(s3_client.s3like, data_files_names)):
-                path_expander = s3_client.glob
-            else:
-                raise ValueError(f"some of s3like links are bad: {data_files_names}")
-    else:
-        data_files_names = expand_braces(files_pattern)
+    if "://" in files_pattern:
+        if not s3_client:
+            raise RuntimeError(s3_client_abscence_error)
+        if all(map(s3_client.s3like, data_files_names)):
+            path_expander = s3_client.glob
+        else:
+            raise ValueError(f"some of s3like links are bad: {data_files_names}")
```

Comment:
If this functionality needs to be used only in certain functions and only under certain conditions, then it is better to move the import inside.

```suggestion
    if "://" in files_pattern:
        from .s3_client import s3_client

        if all(map(s3_client.s3like, data_files_names)):
            path_expander = s3_client.glob
        else:
            raise ValueError(f"some of s3like links are bad: {data_files_names}")
```

Reply:
Corrected. `braceexpand` was also moved.